native/dll/DShowCapture-2/DeviceDiscovery.cpp

```diff
@@ -443,42 +444,78 @@ static int MergeNameList( JNIEnv *env, DEVNAME* DevName, int numDev, DEVNAME* De
+
                 //ZQ hard code for anysee tuner
 				//Anysee hybrid tuners share the same loc (USB) 
 				if ( !strncmp( DevName[j].FriendlyName, "anysee BDA Digital Tuner", 22 ) )
 					j = numDev;
 
-                /* KSF: hard code for Hauppauge WinTV-quadHD, which has 2 BDA Reciever Components 
+
+                /* ----------------
+                 * KSF: hardcode for Hauppauge WinTV-quadHD, which has 2 BDA Reciever Components 
                  * (885 TS Capture, 885 TS Capture 2) for both of it's 2 Video Capture Sources
-                 "  (i.e., at both of it's 2 PCIe hardware locations).
-                 * "Hauppauge WinTV 885 TS Capture" has already been found at this hardware_loc; 
-                 * following lines added for 2nd dev.
-                 * NOTE: comparison here uses index i, not j (unlike anysee BDA, above)
+                 * (i.e., at both of it's 2 PCIe hardware locations).
+                 *
+                 * When 'Hauppauge WinTV 885 Video Capture' was detected (earlier), "Hauppauge WinTV 885 TS Capture' 
+                 * was also entered into DevName[] (due to registry contents).
+                 * If new device is the 2nd TS Capture device, then add it, here.
                  */
                 else if (!strncmp(DevName1[i].FriendlyName, "Hauppauge WinTV 885 TS Capture 2", 32))
                     j = numDev;
+
+
+                /* ----------------
+                 * KSF: hardcode for Hauppauge WinTV-dualHD usb stick, which has a single 'TS Capture' and 
+                 * 2 Source Filters (ATSC Tuner, ATSC Tuner 2) at it's hardware location.
+                 * We don't want user to see (or be able to select) the TS Capture device.
+                 *
+                 * The normal device-detection process finds the following (in this order):
+                 * 1: Hauppauge WinTV-dualHD TS Capture (which would normally tend to bind to 'ATSC Tuner 2')
+                 * 2: Hauppauge WinTV-dualHD ATSC Tuner 2
+                 * 3: Hauppauge WinTV-dualHD ATSC Tuner (not sure why this one is found last; it's first in the current .inf)
+                 *
+                 * Desired result is that DevName[] contains both Tuner names, but not TS Capture 
+                 */
+                else if (!strncmp(DevName[j].FriendlyName, "Hauppauge WinTV-dualHD TS Capture", sizeof("Hauppauge WinTV-dualHD TS Capture"))) 
+                {
+                    // "Hauppauge WinTV-dualHD TS Capture" is present, replace it with the 
+                    //  1st '...WinTV-dualHD ATSC Tuner *' found at same location (generally '...Tuner 2').
+                    slog((env, "patch: replace old DevName[%d] '%s' with new DevName1[%d] '%s' \r\n", 
+                        j, DevName[j].FriendlyName, i, DevName1[i].FriendlyName));
+
+                    strncpy(DevName[j].FriendlyName, DevName1[i].FriendlyName, sizeof(DevName1[i].FriendlyName) - 1);
```

Comment:
Doesn't seem like it makes any difference, since both the src (Devname1) and dest (DevName) are of type DEVNAME.  But I'm willing make the change if you'd prefer.